### ISSUE TITLE
TF IRV Multiple fit() calls and reloading from custom checkpoint

### ISSUE THREAD

**Issue by Malnammi** [deepchem/deepchem#538]
I spent some time going over the code under models/tensorflow_models. I see that it uses TF checkpointing under 'logdir/model.ckpt-#'. Using the models/tensorflow_models/IRV model, I want to do the following:

1. Setup model
2. fit() for 5 epochs.
3. Copy last checkpoint (auto saved after ending training) with name 'custom.ckpt'
4. Call predict_proba(), but I notice here that code reloads the last checkpoint under 'logdir/model.ckpt-#' naming scheme for the eval_graph. Then the code sets self._restored_model = True.
5. fit() for another 5 epochs. The checkpointing scheme will start from 0 again, overwriting checkpoints, which is fine.
6. Now if I call predict_proba(), since self._restored_model == True already, it will not reload the most _last_ checkpoint, but rather use the old eval_graph from 4. 

Am I understanding this correctly? I guess I see two issues here: can't predict_proba with most recent trained model when using multiple fit() calls, and I can't find a way of reloading tf model from a custom checkpoint. I haven't used TF in a while, is there a reason you have the self._restored_model guard? Without it, I can just change the custom checkpoint name to the most recent checkpoint.

**Comment by rbharath** [comment]
I think this summary is accurate. The old `self._restored_model` guard was in pace to handle ops like Dropout that behave differently for train and test. (Not the best solution for handling dropout, but it sufficed for our workflows).

We're currently working on a new Tensorflow backend `TensorGraph`. 

@lilleswing, @peastman  I think the workflow @Malnammi proposed above should work with `TensorGraph`. Is that correct?

@miaecle is working on porting our older models in MoleculeNet to the new backend, but this process will probably take another couple months.

I think this issue would likely be tricky to fix in `TensorflowModel` since it requires large semantic changes and we're in the process of deprecating that code. I'd be very open to a PR porting the IRV model over to `TensorGraph` though. Would solve the problem in a principled fashion and be very helpful :)

**Comment by Malnammi** [comment]
Thanks for the reply. I went over the code under models/tensorgraph. It looks like there is support for loading from disk with `load_from_dir(model_dir)`. 

I might be able to port the IRV code over the weekend which I believe should be in the same spirit as tensorgraph/models/robust_multitask.py. So just use a MultiTaskTensorGraph and build the IRV network for each task, seems about right?

**Comment by rbharath** [comment]
Great! I think MultiTaskTensorGraph might be obsolete. You should be able to just use TensorGraph directly.

**Comment by rbharath** [comment]
I believe IRV has been posted to TensorGraph so closing this issue. Feel free to re-open if necessary.
